perfkitbenchmarker/vm_util.py

```diff
@@ -501,3 +504,51 @@ def GenerateSSHConfig(vms):
     ofp.write(template.render({'vms': vms}))
   logging.info('ssh to VMs in this benchmark by name with: '
                'ssh -F {0} <vm name>'.format(target_file))
+
+
+@contextlib.contextmanager
+def RunDStatIfConfigured(vms):
+  """Installs and runs dstat on 'vms' if FLAGS.dstat is set.
+
+  On exiting the context manager, the results are copied to the run temp dir.
+
+  Args:
+    vms: List of virtual machines.
+
+  Yields:
+    None
+  """
+  if not FLAGS.dstat:
+    yield
+    return
+
+  lock = threading.Lock()
+  pids = {}
+  file_names = {}
+
+  def StartDStat(vm):
+    vm.Install('dstat')
+    dstat_file = os.path.join(VM_TMP_DIR, '{0}-dstat.csv'.format(vm.name))
```

Comment:
@ehankland: done, thanks.
@voellm: one dstat file per VM. You're right, they would be overridden if multiple benchmarks are specified to `--benchmarks`. I can add a uuid, or we could tag each file with the benchmark run. Any preference?